v1.12 Backports 2023-08-23

tklauser opened the pull request:
* [ ] #27495 (@ishuar)
 * [x] #27409 (@tanjunchen)
   * :warning: minor conflict, resolved by omitting changes to non-existent files
   * (tam) I have verified this backport commit.
 * [x] #27457 (@brlbil)
   * :warning: minor conflicts
 * [ ] #27537 (@qmonnet)
 * [x] ~#27532 (@doniacld)~ already backported by author
 * [x] #27511 (@rolinh)
   * :warning: minor conflicts in files generated from helm values. Resolved by regenerating
 * [x] #27597 (@sayboras)
   * :warning: conflict in `install/kubernetes/cilium/templates/validate.yaml`,skipped 2nd commit per https://github.com/cilium/cilium/pull/27660#issuecomment-1689981684

PRs skipped due to conflicts:

 * #27346 (@qmonnet)

Once this PR is merged, you can update the PR labels via:
```upstream-prs
for pr in 27495 27409 27457 27537 27511 27597; do contrib/backporting/set-labels.py $pr done 1.12; done
```
or with
```
make add-labels BRANCH=v1.12 ISSUES=27495,27409,27457,27537,27511,27597
```

brlbil committed:
ci: update tested k8s version for aks

[ upstream commit 4cb2a4b4e2e0c0026c065905f2da41bd1acfc4e4 ]

AKS supported k8s versions changed.
This commit updates tested k8s versions for AKS

Signed-off-by: Birol Bilgin <birol@cilium.io>
Signed-off-by: Tobias Klauser <tklauser@distanz.ch>

tanjunchen committed:
update L7 traffic CiliumClusterwideEnvoyConfig example

[ upstream commit 0cd6932b45d5f33824d6e2a739441152b4c67785 ]

Signed-off-by: chentanjun <tanjunchen20@gmail.com>
Signed-off-by: Tobias Klauser <tklauser@distanz.ch>

ishuar committed:
fix makefile sed command to be compatible with macOs

[ upstream commit b7947bc216e21d9c9813609b7bb621fbfa32d052 ]

Signed-off-by: ishuar <ishansharma887@gmail.com>
Signed-off-by: Tobias Klauser <tklauser@distanz.ch>

rolinh committed:
update cilium/certgen to v0.1.9

[ upstream commit 164332f95c9e2b1b6e55c2db42b701d591a93fb8 ]

v0.1.9 is the first release of certgen 1.5 years. As a patch release, it
doesn't contain any major changes but only updates to its dependencies
and the Go version that it's built with.

See release notes for details:
https://github.com/cilium/certgen/releases/tag/v0.1.9

Signed-off-by: Robin Hahling <robin.hahling@gw-computing.net>
Signed-off-by: Tobias Klauser <tklauser@distanz.ch>

qmonnet committed:
docs: Fix config option for spelling filters

[ upstream commit 270be303aa13ca9c1441faf0fbe602045ee9dca1 ]

We've been using a custom spelling filter to make sure that "WireGuard"
is spelled correctly, with the proper case. As it turns out, passing
this filter to the Sphinx configuration, in the conf.py file, makes
Sphinx re-read all sources and re-write all output files, as can be
observed when running sphinx-build multiple times, without suppressing
the output:

    $ sphinx-build -M html . _build
    [...]
    updating environment: [config changed ('spelling_filters')] 472 added, 30 changed, 0 removed
    [...]

This is because in conf.py, we pass the filter directly as a function.
When Sphinx writes its environment.pickle file to keep track of the
configuration in use, it discards values that cannot be serialised,
including the filter, "<class 'cilium_spellfilters.WireGuardFilter'>",
of instance "type" [0]. So the value for the configuration option
"spelling_filters" is not saved, and as Sphinx believes that the
configuration has changed, it reads and rebuilds everything.

In fact, the issue has been reported before, and solved in the
spellchecker [1]. We need to set the configuration with a string instead
of the direct function object, and the extension is able to load it from
there. Let's adjust accordingly, to save cycles when building the docs
more than once.

[0] https://github.com/sphinx-doc/sphinx/blob/v7.1.2/sphinx/config.py#L323
[1] https://github.com/sphinx-contrib/spelling/pull/40

Signed-off-by: Quentin Monnet <quentin@isovalent.com>
Signed-off-by: Tobias Klauser <tklauser@distanz.ch>

sayboras reviewed (approved):
Thanks and looks good for my changes.

sayboras reviewed (approved):
Thanks :heavy_check_mark:

sayboras committed:
proxy: Ignore visibility annotation if proxy is disabled

[ upstream commit fe375869886bad52df9a1e7ccfee692dd77d7452 ]

This is to avoid the below error if proxy is disabled as part of
installation, but the visibility annotation is added to pods later.

```
2023-08-11T12:42:41.390575371Z goroutine 1522 [running]:
2023-08-11T12:42:41.390581994Z github.com/cilium/cilium/pkg/proxy.(*Proxy).CreateOrUpdateRedirect(0x0, {0x3ae74b8, 0xc000650280}, {0x3aeb440?, 0xc0016e4560?}, {0xc0010ba1b0, 0x12}, {0x3afd260, 0xc000982000}, 0xc001c9c980)
2023-08-11T12:42:41.390589198Z  github.com/cilium/cilium/pkg/proxy/proxy.go:459 +0xb7
2023-08-11T12:42:41.390596081Z github.com/cilium/cilium/pkg/endpoint.(*Endpoint).addVisibilityRedirects(0xc000982000, 0x1, 0xc001874ba0?, 0xc001c9c980?)
2023-08-11T12:42:41.390602613Z  github.com/cilium/cilium/pkg/endpoint/bpf.go:343 +0x443
2023-08-11T12:42:41.390614345Z github.com/cilium/cilium/pkg/endpoint.(*Endpoint).addNewRedirects(0xc000982000, 0xc001874870?)
2023-08-11T12:42:41.390651325Z  github.com/cilium/cilium/pkg/endpoint/bpf.go:424 +0x3c5
2023-08-11T12:42:41.390658068Z github.com/cilium/cilium/pkg/endpoint.(*Endpoint).runPreCompilationSteps(0xc000982000, 0xc000e31800, 0x0)
2023-08-11T12:42:41.390663999Z  github.com/cilium/cilium/pkg/endpoint/bpf.go:802 +0x4fe
2023-08-11T12:42:41.390669690Z github.com/cilium/cilium/pkg/endpoint.(*Endpoint).regenerateBPF(0xc000982000, 0xc000e31800)
2023-08-11T12:42:41.390675451Z  github.com/cilium/cilium/pkg/endpoint/bpf.go:542 +0x1a5
2023-08-11T12:42:41.390681112Z github.com/cilium/cilium/pkg/endpoint.(*Endpoint).regenerate(0xc000982000, 0xc000e31800)
2023-08-11T12:42:41.390686893Z  github.com/cilium/cilium/pkg/endpoint/policy.go:467 +0x9c6
2023-08-11T12:42:41.390692564Z github.com/cilium/cilium/pkg/endpoint.(*EndpointRegenerationEvent).Handle(0xc000131b50, 0x0?)
2023-08-11T12:42:41.390698385Z  github.com/cilium/cilium/pkg/endpoint/events.go:53 +0x325
2023-08-11T12:42:41.390704045Z github.com/cilium/cilium/pkg/eventqueue.(*EventQueue).run.func1()
2023-08-11T12:42:41.390709716Z  github.com/cilium/cilium/pkg/eventqueue/eventqueue.go:245 +0x142
```

Fixes: #27594

Signed-off-by: Tam Mach <tam.mach@cilium.io>
Signed-off-by: Tobias Klauser <tklauser@distanz.ch>

joestringer reviewed (approved):
Ack for the last couple of PRs.
